lib/ci.js

```diff
@@ -1,31 +1,91 @@
 #!/usr/bin/env node
 
-'use strict';
+"use strict";
 
-var existsSync = require('fs').existsSync;
-var spawnSync = require('child_process').spawnSync;
-var getPnpmVersion = require('./get-pnpm-version');
+var existsSync = require("fs").existsSync;
+var spawnSync = require("child_process").spawnSync;
+var getPnpmVersion = require("./get-pnpm-version");
 
 var opts = {
-	stdio: 'inherit',
+	stdio: "inherit",
 	shell: true,
 };
 
 var spawned;
 
-if (existsSync('package-lock.json')) {
-	spawned = spawnSync('npm', ['ci'], opts);
-} else if (existsSync('yarn.lock')) {
-	spawned = spawnSync('npx', ['yarn', 'install', '--immutable'], opts);
-} else if (existsSync('pnpm-lock.yaml')) {
-	var nodeVersion = process.version.slice(1).split('.').map(function (v) {
-		return parseInt(v, 10);
-	});
-	
+if (!existsSync("package.json")) {
+	console.error("No `package.json` found");
+	process.exit(1);
+}
```

Comment:
This would be a breaking change, and an unnecessary one.

Move `existsSync` check to L23

Reply:
If no `package.json` found, `npm`, `yarn` and `pnpm` should already be broken?

Reply:
Although I disagree with your comment, but it is just simple and as you said "unnecessary", so I've just refactored.